lib/std/Progress.zig

```diff
@@ -419,6 +419,17 @@ pub fn start(options: Options) Node {
                 };
             }
 
+            std.io.tty.setAbortSignalHandler(struct {
+                fn abort_signal_handler() void {
+                    switch (global_progress.terminal_mode) {
+                        .off => unreachable, // handled a few lines above
+                        .ansi_escape_codes => clearWrittenWithEscapeCodes() catch {},
+                        .windows_api => if (is_windows) clearWrittenWindowsApi() catch {} else unreachable,
+                    }
+                    std.process.exit(0);
```

Comment:
This changes behavior slightly.
Before:
```
~/zig@abort$ zig-out/bin/zig test ../x.zig --zig-lib-dir lib
^C63] AST Lowering
~/zig@abort$ echo $?g
130
~/zig@abort$ otocol.zig
```
After:
```
~/zig@abort$ zig-out/bin/zig test ../x.zig --zig-lib-dir lib
^C~/zig@abort$ echo $?
0
~/zig@abort$
```
As you can see it seems that by default forcibly exiting a program using Ctrl+C causes it to exit with an exit code of 130. But now it's exiting with 0 because we're exiting manually at our own will.

I don't know how much this applies and in which other cases it might exit with an exit code other than 0 or 130 (also thinking about Windows).

I don't think this is a problem. Just pointing it out.

Reply:
That'd lead to incorrect behavior in the following cases:
 - `zig build && ./zig-out/bin/myapp`
 - shell scripts

[Some relevant info.](https://www.cons.org/cracauer/sigint.html)

Reply:
> You cannot "fake" the proper exit status by an exit(3) with a special numeric value, even if you look up the numeric value for your system.

Do you think at least exiting with a non-zero exit code would be better? Any suggestions?

Reply:
I'd look into `std.process.abort()`

Reply:
I tried propagating the signal by reraising it using `std.posix.raise(std.posix.SIG.INT)` but that simply reinvokes the same signal handler.

`std.process.abort()` seems to exit with `134` on my system but prints `^CAborted` which is undesirable so for now I'm going with `std.process.exit(1)` which I think is a perfectly fine exit code in any case because the program was interrupted so it had no chance to succeed, so the result is failure indicated by exit code 1.